### PR TITLE
[Video Capture] Audio alerts for recording timer

### PR DESCRIPTION
**Pull request by manikd31**
## AV-3983: Audible Alerts for Video-Capture script
- Ticket: [AV-3983](https://20bn.atlassian.net/browse/AV-3983)

### Developer Notes:
This PR aims at adding audible alerts to the `video_capture.py` script. In addition to the visual timer for countdown, audio-alerts will be played in both, pre-recording and recording phases with the following conditions:
- **Pre-recording**: Count down (alert) the last three seconds of the timer, and another alert for pre-record phase end, i.e. start recording.
- **Recording**: Just a single alert when the recording phase has ended.

**Usage**: Audio-alerts are by default enabled. To disable these, add a flag for `--no_audio` while calling `video_capture.py`

**Edit (from Annotation)**: Changed text thickness on video-prompt to be more visible to the user

### CODE REVIEW

File: scripts/tools/video_capture.py [hunk 1]
```diff
@@ -25,9 +25,43 @@
...
+
+
+class ShutDownWatcher(Thread):
+    def __init__(self, shutdown_event, shutdown_fn=None):
+        Thread.__init__(self)
+        self.shutdown_event = shutdown_event
+        self.shutdown_fn = shutdown_fn
+
+    def run(self):
+        print("THIS IS A DEBUG PRINT TO SEE IF RUN IS BEING CALLED SUCCESSFULLY")
```
Comment (sunny-panchal) [inline scripts/tools/video_capture.py:45]: leftover

File: scripts/tools/video_capture.py [hunk 2]
```diff
@@ -25,9 +25,43 @@
 
 import cv2
 from docopt import docopt
+from pathlib import Path
+import simpleaudio as sa
+from threading import Thread
 
 FONT = cv2.FONT_HERSHEY_PLAIN
 _shutdown = False
+countdown_sound = 'counter_sound.wav'
```
Comment (sunny-panchal) [inline scripts/tools/video_capture.py:34]: constant should be all caps

File: scripts/tools/video_capture.py [hunk 3]
```diff
@@ -46,24 +80,32 @@ def _capture_video(video_duration=0., record=False):
...
+                countdown -= 1
+
             ret, frame_norm = cap.read()
             frame = cv2.flip(frame_norm, 1)
             frames.append(frame.copy())
             frame_size = (frame.shape[1], frame.shape[0])
 
             if record:
-                message = f"recording video {str(index + 1)}"
+                message = f"Recording Video {str(index + 1)}"
```
Comment (sunny-panchal) [inline scripts/tools/video_capture.py:98]:
```suggestion
                message = f"Recording video {str(index + 1)}"
```

**Review by sunny-panchal — COMMENTED** [review]
Great! It works but I think there are still a few things to address:
- add required packages to `requirements.txt`
- refactor media files to either:
    - `docs/audio/...`
    - `scripts/tools/video_capture/...`
    - The tones play over "2, 1, 0, end." Should it be "3, 2, 1, end"? displayed countdown values should be incremented by 1
    - I think the tone for countdown and done sound better swapped

File: scripts/tools/video_capture.py [hunk 4]
```diff
@@ -25,9 +25,43 @@
...
+import simpleaudio as sa
+from threading import Thread
 
 FONT = cv2.FONT_HERSHEY_PLAIN
 _shutdown = False
+countdown_sound = 'counter_sound.wav'
+done_sound = 'long_done_sound.wav'
+
+
+class ShutDownWatcher(Thread):
```
Comment (sunny-panchal) [inline scripts/tools/video_capture.py:38]: Also, this class is unused

File: scripts/tools/video_capture.py (hunk 1 above)
Comment (manikd31) [reply scripts/tools/video_capture.py]: Removed this class, was not being used.

File: scripts/tools/video_capture.py (hunk 2 above)
Comment (manikd31) [reply scripts/tools/video_capture.py]: Done.

File: scripts/tools/video_capture.py (hunk 3 above)
Comment (manikd31) [reply scripts/tools/video_capture.py]: Done.

File: scripts/tools/video_capture.py (hunk 4 above)
Comment (manikd31) [reply scripts/tools/video_capture.py]: Removed.

**Comment by manikd31** [comment]
> Great! It works but I think there are still a few things to address:
> 
>     * add required packages to `requirements.txt`
> 
>     * refactor media files to either:
>       
>       * `docs/audio/...`
>       * `scripts/tools/video_capture/...`
>       * The tones play over "2, 1, 0, end." Should it be "3, 2, 1, end"? displayed countdown values should be incremented by 1
>       * I think the tone for countdown and done sound better swapped

Addressed all comments and suggestions, apart from swapping `done` and `countdown` sounds. Do we need more opinions on that? I personally feel having a high-pitch sounds for `Done` makes sense, but we can swap that anytime.

**Comment by manikd31** [comment]
> > Great! It works but I think there are still a few things to address:
> > ```
> > * add required packages to `requirements.txt`
> > 
> > * refactor media files to either:
> >   
> >   * `docs/audio/...`
> >   * `scripts/tools/video_capture/...`
> >   * The tones play over "2, 1, 0, end." Should it be "3, 2, 1, end"? displayed countdown values should be incremented by 1
> >   * I think the tone for countdown and done sound better swapped
> > ```
> 
> Addressed all comments and suggestions, apart from swapping `done` and `countdown` sounds. Do we need more opinions on that? I personally feel having a high-pitch sounds for `Done` makes sense, but we can swap that anytime.

Updated sounds for `done` and `countdown`. Also, added another sound for `exit`.

**Review by sunny-panchal — APPROVED** [review]
Had to increase the margin otherwise the countdown wouldn't play on a few trials otherwise looks good :+1:

**Comment by florianletsch** [comment]
@sunny-panchal @manikd31 Where are the audio files from? What's the license? We have to include the appropriate license disclaimer in the README because binary files are usually not licensed under MIT.

**Comment by sunny-panchal** [comment]
> @sunny-panchal @manikd31 Where are the audio files from? What's the license? We have to include the appropriate license disclaimer in the README because binary files are usually not licensed under MIT.

Good point... @manikd31 is rejoining us tomorrow and I'll confirm with him.

**Pull request merged**